<launch>
    <arg name="manager_name" default="nodelet_manager" />
    <node pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen" >
        <param name="num_worker_threads" value="8" />
    </node>

        <!--定义了一个名为 "galaxy_camera" 的节点-->
<!--    <node pkg="nodelet" type="nodelet" name="galaxy_camera"-->
<!--          args="load galaxy_camera/galaxy_camera_nodelet $(arg manager_name)"-->
<!--          output="screen">-->
<!--&lt;!&ndash;         <param name="camera_SN" value="KE0200060392" />&ndash;&gt;-->
<!--    </node>-->

<!--    定义了一个名为 "hk_camera" 的节点-->
     <node pkg="nodelet" type="nodelet" name="hk_camera"
           args="load hk_camera/HKCameraNodelet $(arg manager_name)"
           output="screen">
         <param name="camera_name" value="hk_camera"/>
         <param name="image_width" value="1440"/>
         <param name="image_height" value="1080"/>
         <param name="pixel_format" value="bgr8"/>
         <param name="camera_info_url" value=""/>
         <param name="camera_frame_id" value="camera_optical_frame"/>
         <param name="camera_sn" value="00J67375789"/>
         <param name="frame_rate" value="15"/>
     </node>

<!--    <node pkg="nodelet" type="nodelet" name="Processor"-->
<!--          args="load rm_digtialimg_proc_deep/Processor $(arg manager_name)" >-->

<!--&lt;!&ndash;        <remap from="/hk_camera/image_raw" to="/galaxy_camera/galaxy_camera/image_raw"/>&ndash;&gt;-->
<!--        <remap from="/hk_camera/image_raw" to="/hk_camera/image_raw"/>-->
<!--    </node>-->

    <node pkg="nodelet" type="nodelet" name="processor"
          args="load rm_digtialimg_proc_deep/Processor $(arg manager_name)" > <!--加载了名为 "Processor" 的插件，并设置了一些参数-->
        <rosparam file="$(find rm_digtialimg_proc_deep)/config/weibull_coefficient.yaml" command="load" />
        <rosparam file="$(find rm_digtialimg_proc_deep)/config/armor_coefficient.yaml" command="load" />
        <rosparam file="$(find rm_digtialimg_proc_deep)/config/preprocess_coefficient.yaml" command="load" />
        <rosparam file="$(find rm_digtialimg_proc_deep)/config/inference.yaml" command="load" />
        <remap from="/processor/result_msg" to="/processor/result_msg"/>
    </node>


<!--    &lt;!&ndash; 启动视频检测节点 &ndash;&gt;-->
<!--    <node name="video_detector_node" pkg="rm_digtialimg_proc_deep" type="video_detector_node"  output="screen"/>-->

</launch>